jax/_src/scipy/signal.py

```diff
@@ -146,3 +152,343 @@ def detrend(data, axis=-1, type='linear', bp=0, overwrite_data=None):
       coef, *_ = linalg.lstsq(A, data[sl])
       data = data.at[sl].add(-jnp.matmul(A, coef, precision=lax.Precision.HIGHEST))
     return jnp.moveaxis(data.reshape(shape), 0, axis)
+
+
+def _fft_helper(x, win, detrend_func, nperseg, noverlap, nfft, sides):
+  """Calculate windowed FFT in the same way the original SciPy does.
+  """
+  if x.dtype.kind == 'i':
+    x = x.astype(win.dtype)
+
+  # Created strided array of data segments
+  if nperseg == 1 and noverlap == 0:
+    result = x[..., np.newaxis]
```

Comment:
FYI: This leaves batch_shape undefined, resulting in an exception at line 181.

Reply:
Thanks for reporting. Fixed.